reco_utils/common/tf_utils.py

```diff
@@ -1,10 +1,16 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
-import abc
+import collections
 import itertools
-import tensorflow as tf
+
 import numpy as np
 import pandas as pd
+import tensorflow as tf
+from tensorflow.python.estimator.inputs.queues import feeding_functions
+from tensorflow.python.estimator.inputs.numpy_io import (
+    _get_unique_target_key,
```

Comment:
this hurts a little bit lol, but I guess it's not our fault. I guess there are no public functions that we can use right?

Reply:
ah... yeah. Not sure why they did not expose the seed variable to numpy input function.
If we really don't want to do this, alternative way is to shuffle by ourselves like:

```
def numpy_input_fn(x, ...):
    ...
    if shuffle==True and seed is not None:
        # set random seed and shuffle x
        ....
        # then call tf's numpy_input_fn with the shuffled x
        tf.estimator.inputs.numpy_input_fn(x)
```

I actually like this solution. Much cleaner.

Reply:
Ah... the thing is, w/ this, we cannot shuffle for every epoch... hmmm

Reply:
I think ‘tf.set_random_seed’ will work for this case. Let me try and will update codes if works.

Reply:
Refactored to use `tf.data.dataset` (newer API) within our `pandas_input_fn` and now much cleaner.